tasks/run_cd4pe_job.rb

```diff
@@ -403,17 +413,24 @@ def make_working_dir(working_dir)
     base_64_ca_cert = params['base_64_ca_cert']
 
     set_job_env_vars(params)
-    make_working_dir(working_dir)
 
-    job_runner = CD4PEJobRunner.new(working_dir: working_dir, docker_image: docker_image, docker_run_args: docker_run_args, job_token: job_token, web_ui_endpoint: web_ui_endpoint, job_owner: job_owner, job_instance_id: job_instance_id, base_64_ca_cert: base_64_ca_cert, windows_job: windows_job, logger: @logger)
+    root_job_dir = File.join(Dir.pwd, 'cd4pe_job_working_dir')
+    make_dir(root_job_dir)
+    @working_dir = File.join(root_job_dir, "cd4pe_job_instance_#{job_instance_id}_#{DateTime.now.strftime('%Q')}")
+    make_dir(@working_dir)
+
+    job_runner = CD4PEJobRunner.new(working_dir: @working_dir, docker_image: docker_image, docker_run_args: docker_run_args, job_token: job_token, web_ui_endpoint: web_ui_endpoint, job_owner: job_owner, job_instance_id: job_instance_id, base_64_ca_cert: base_64_ca_cert, windows_job: windows_job, logger: @logger)
     job_runner.get_job_script_and_control_repo
     output = job_runner.run_job
 
     output[:logs] = @logger.get_logs
     puts output.to_json
 
+    delete_dir(@working_dir)
+    
     exit get_combined_exit_code(output)
   rescue => e
+    delete_dir(@working_dir)
```

Comment:
Instead of putting it in both the `try` and `rescue` blocks, just put it once in a new `ensure` block?